R/map_estimate.R

```diff
@@ -57,10 +57,27 @@ map_estimate.numeric <- function(x, precision = 2^10, method = "kernel", ...) {
 }
 
 .map_estimate <- function(x, precision = 2^10, method = "kernel", ...) {
-  d <- estimate_density(x, precision = precision, method = method, ...)
-
-  out <- d$x[which.max(d$y)]
-  attr(out, "MAP_density") <- max(d$y)
+  dots <- list(...)
+  # sanity check - if we have only one unique value (a vector of constant values)
+  # density estimation doesn't work
+  if (insight::n_unique(x) == 1) {
+    out <- stats::na.omit(x)[1]
+    attr(out, "MAP_density") <- 1
+  } else {
+    d <- try(estimate_density(x, precision = precision, method = method, ...), silent = TRUE)
+    if (inherits(d, "try-error")) {
+      if (isTRUE(dots$verbose)) {
+        msg <- "Could not calculate MAP estimate."
+        if (grepl("too sparse", d, fixed = TRUE)) {
```

Comment:
Data is _maybe_ too sparse gets `NA` + warning,
but singular data (i.e., data that _is_ too sparse) gets singular value and no warning. This feels inconsistent. I this we should add a similar warning (that can be turned off by `varbose=`) for the singular case: "The provided data is a constant. Returning constant."